processor/src/test/java/org/mapstruct/ap/test/factories/Bar6Factory.java

```diff
@@ -0,0 +1,33 @@
+/**
+ *  Copyright 2012-2016 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.test.factories;
+
+import org.mapstruct.ObjectFactory;
+
+/**
+ * @author Remo Meier
+ */
+public class Bar6Factory {
+
+    @ObjectFactory
+    public Bar6 createBar6(Foo6A foo6A) {
```

Comment:
This wouldn't work if you would require only `Foo6B` as argument type, right?

I think somewhere around the Lifecycle methods, we have some code that finds all possible (assignable) combinations for passing source/target arguments to all candidate methods. I think we need to do something very similar here (maybe even reuse the code?), but raise an error in case of multiple matches - at least I can't say from the top of my head how we could select a "best matching" one in case of ambiguities. Well, perhaps favor a factory method with arguments above one without any arguments. But other than that, how would we choose between one that takes Foo6A and one that takes Foo6B?